internal/pkg/scaffold/olm-catalog/descriptor/parse.go

```diff
@@ -0,0 +1,261 @@
+				}
+			default:
+				return parsedCRDDescriptions{}, errors.Errorf("unsupported %s child path element %s", parentPathElem, childPathElems[0])
+			}
+		default:
+			return parsedCRDDescriptions{}, errors.Errorf("unsupported path element %s", parentPathElem)
+		}
+	}
+	pd.descriptors = append(pd.descriptors, specd, statusd)
+	return pd, nil
+}
+
+// parseMemberAnnotation determines which descriptor annotation was passed from
+// pathElems and sets val to the corresponding field in d.
+func parseMemberAnnotation(d *descriptor, pathElems []string, val string) (err error) {
+	switch len(pathElems) {
+	case 1:
+		// If this case is never entered, d will not be included.
+		d.include, err = strconv.ParseBool(val)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing %s bool val %s", pathElems[0], val)
+		}
+	case 2:
+		switch pathElems[1] {
+		case "displayName":
+			d.DisplayName, err = strconv.Unquote(val)
+			if err != nil {
+				return errors.Wrapf(err, "error unquoting %s", val)
+			}
+		case "x-descriptors":
+			xdStr, err := strconv.Unquote(val)
+			if err != nil {
+				return errors.Wrapf(err, "error unquoting %s", val)
+			}
+			d.XDescriptors = strings.Split(xdStr, ",")
+		default:
+			return errors.Errorf("unsupported descriptor path element %s", pathElems[1])
+		}
+	default:
+		return errors.Errorf("unsupported descriptor path %s", annotations.JoinPath(pathElems...))
+	}
+	return nil
+}
+
+// parseResource parses a resource string of the form:
+// "kind,version,\"quoted name\""
+func parseResource(rStr string) (r olmapiv1alpha1.APIResourceReference, err error) {
+	rStr, err = strconv.Unquote(rStr)
+	if err != nil {
+		return r, err
+	}
+	rSplit := strings.SplitN(rStr, ",", 3)
+	if len(rSplit) < 2 {
+		return r, errors.Errorf("resource string %s did not have at least a kind and a version", rStr)
+	}
+	r.Kind, r.Version = strings.TrimSpace(rSplit[0]), strings.TrimSpace(rSplit[1])
+	if len(rSplit) == 3 {
+		r.Name, err = strconv.Unquote(rSplit[2])
+		if err != nil {
+			return r, err
+		}
+		r.Name = strings.TrimSpace(r.Name)
+	}
+	return r, nil
+}
+
+// parseDescription joins comment strings into one line, removing any tool
+// directives.
+func parseDescription(comments []string) string {
+	var lines []string
+	for _, c := range comments {
+		l := strings.TrimSpace(strings.TrimLeft(c, "/"))
+		if l == "" || strings.HasPrefix(l, "+") {
+			continue
+		}
+		lines = append(lines, l)
+	}
+	return strings.Join(lines, " ")
+}
+
+var jsonTagRe = regexp.MustCompile(`json:"([a-zA-Z0-9,]+)"`)
+
+func parsePathFromJSONTags(tags string) string {
+	tagMatches := jsonTagRe.FindStringSubmatch(tags)
+	if len(tagMatches) > 1 {
+		ts := strings.Split(tagMatches[1], ",")
+		if len(ts) != 0 && ts[0] != "" {
+			return ts[0]
+		}
+	}
+	return ""
+}
+
+// From https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L5-L14
+var specXDescriptors = map[string]string{
```

Comment:
```suggestion
const specXDescriptors = map[string]string{
```

Reply:
Go does not support [const maps](https://groups.google.com/forum/#!topic/golang-nuts/xUaDMciyB5Q)

Reply:
👍